ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -198,14 +197,17 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
   }
 
   /**
-   * Filters deprecated {@link Container}s for compaction purpose. Deprecated status include DELETE_IN_PROGRESS and INACTIVE.
+   * Filters deprecated {@link Container}s for compaction purpose. Deprecated containers include DELETE_IN_PROGRESS
+   * containers met with retention time and all INACTIVE containers.
    * @return the deprecated {@link Container}s' accountId & containerId pairs.
    */
   private void getDeprecatedContainers() {
     deprecatedContainers.clear();
     if (accountService != null) {
       accountService.getContainersByStatus(Container.ContainerStatus.DELETE_IN_PROGRESS).forEach((container) -> {
-        deprecatedContainers.add(new Pair<>(container.getParentAccountId(), container.getId()));
+        if (container.getLastUpdateTime() + TimeUnit.DAYS.toMillis(14) >= System.currentTimeMillis()) {
```

Comment:
nit: we should use a config for this value, maybe we can update it in next pr.

Reply:
Updated.